lib/absinthe/schema/notation.ex

```diff
@@ -33,6 +33,18 @@ defmodule Absinthe.Schema.Notation do
     end
   end
 
+  defmacro topic(topic_fun) do
+    __CALLER__.module
+    |> Scope.put_attribute(:topic, topic_fun)
+    :ok
+  end
+
+  defmacro trigger(mutations, attrs) do
+    env = __CALLER__
```

Comment:
Yup. You may want to wait on reviewing until it's closer to being done, this is very WIP, there will undoubtedly by non idiomatic code from time to time until it's closer to completion.